src/curl.cpp

```diff
@@ -1856,7 +1862,7 @@ S3fsCurl::S3fsCurl(bool ahbe) :
     LastResponseCode(S3FSCURL_RESPONSECODE_NOTSET), postdata(NULL), postdata_remaining(0), is_use_ahbe(ahbe),
     retry_count(0), b_infile(NULL), b_postdata(NULL), b_postdata_remaining(0), b_partdata_startpos(0), b_partdata_size(0),
     b_ssekey_pos(-1), b_ssetype(sse_type_t::SSE_DISABLE),
-    sem(NULL), completed_tids_lock(NULL), completed_tids(NULL), fpLazySetup(NULL)
+    sem(NULL), completed_tids_lock(NULL), completed_tids(NULL), fpLazySetup(NULL), totalwrite(0)
```

Comment:
Instead of tracking `totalwrite`, maybe it is better to store the original `startpos` and `size` as consts which `ResetOffset` can use?